From d4500078e1a747c2288273d1d3b90079da2a2857 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Mon, 25 Nov 2019 12:58:10 -0600
Subject: [PATCH 2/3] opkg.py/get_file_list(): Add xz support

Open data.tar.xz if gz payload is not present. This change allows xz
payloads to be archived in filelist.

However, only python3's tarfile has xz support. This therefore only
works if opkg-make-index's shebang is overridden to run in python3.

Signed-off-by: Haris Okanovic <haris.okanovic@ni.com>

Upstream-Status: Submitted
---
 opkg.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/opkg.py b/opkg.py
index ddb2c38..23fed96 100644
--- a/opkg.py
+++ b/opkg.py
@@ -408,8 +408,12 @@ class Package(object):
             return []
         f = open(self.fn, "rb")
         ar = arfile.ArFile(f, self.fn)
-        tarStream = ar.open("data.tar.gz")
-        tarf = tarfile.open("data.tar.gz", "r", tarStream)
+        try:
+            tarStream = ar.open("data.tar.gz")
+            tarf = tarfile.open("data.tar.gz", "r", tarStream)
+        except IOError:
+            tarStream = ar.open("data.tar.xz")
+            tarf = tarfile.open("data.tar.xz", "r:xz", tarStream)
         self.file_list = tarf.getnames()
         self.file_list = [["./", ""][a.startswith("./")] + a for a in self.file_list]
 
-- 
2.24.0

